torch/distributed/_tensor/op_schema.py

```diff
@@ -54,9 +55,13 @@ class PlacementStrategy:
     """
     A placement strategy describes an acceptable sharding placements of the output
     and the tensor arguments of an operation.
+
+    note: when the op return value is a single DTensor object, output_spec is
+    DTensorSpec; when the return value is a sequence of Optional[DTensor],
+    output_spec is a sequence of Optional[DTensorSpec].
     """
 
-    output_spec: DTensorSpec
+    output_spec: Union[DTensorSpec, Tuple[Optional[DTensorSpec], ...]]
```

Comment:
No need to do in this PR, I feel maybe as a follow up PR, we can refactor this to a cleaner way:

1. rename `output_spec` to `output_specs` to indicate that there can be multiple output returns
2. rename `out_spec` cached property to `output_spec` to align with `input_specs`

Reply:
#116437